localstack/services/s3/s3_starter.py

```diff
@@ -222,3 +217,22 @@ def s3_bucket_response_delete_keys(self, request, body, bucket_name):
 
     s3_responses.S3ResponseInstance._bucket_response_delete_keys = types.MethodType(
         s3_bucket_response_delete_keys, s3_responses.S3ResponseInstance)
+
+    # Patch _handle_range_header(..)
+    # https://github.com/localstack/localstack/issues/2146
+    s3_response_handle_range_header_orig = s3_responses.S3ResponseInstance._handle_range_header
+
+    def s3_response_handle_range_header(self, request, headers, response_content):
+        rs_code, rs_headers, rs_content = s3_response_handle_range_header_orig(request, headers, response_content)
+        if rs_code == 206:
+            for k in ['ETag', 'last-modified']:
+                v = headers.get(k)
+                if v:
```

Comment:
Can we add a check that the header is not yet present in `rs_headers`, before overwriting it:
```
if v and not rs_headers.get(k):
    ...
```